src/webgl/p5.RendererGL.js

```diff
@@ -1970,6 +2083,31 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
     fillShader.bindTextures();
   }
 
+  // getting called from _setFillUniforms
+  _setImageLightUniforms(shader){
+    //set uniform values
+    if( this.activeImageLight == null ){
+      console.log('activeImageLight prop is null');
+    }
+    shader.setUniform('uUseImageLight', this.activeImageLight != null );
+    // true
+    if (this.activeImageLight) {
+      // this.activeImageLight has image as a key
+      // look up the texture from the blurryTexture map
+      // change the name of shader of diffused to prevent confusion
+      let diffusedLight = this.getBlurryTexture(this.activeImageLight);
+      // change the name of this uniform as diff for diffused and specular
+      shader.setUniform('environmentMapDiffused', diffusedLight);
+      let specularLight = this.getSpecularTexture(this.activeImageLight);
+      // shine >= 1 so
+      // 0-1*8 = 0-8
+      // 0-8 * 20 = 0-160
+      let roughness = 20/this._useShininess;
+      shader.setUniform('levelOfDetail', roughness*8);
```

Comment:
We should mention that the 8 here is because we have 8 levels in the mipmap

Reply:
done